packages/repository/src/relations/has-many/has-many-through.repository.ts

```diff
@@ -238,21 +235,38 @@ export class DefaultHasManyThroughRepository<
   }
 
   async link(
-    targetModelId: TargetID,
+    targetId: TargetID,
     options?: Options & {
       throughData?: DataObject<ThroughEntity>;
       throughOptions?: Options;
     },
-  ): Promise<TargetEntity> {
-    throw new Error('Method not implemented.');
+  ): Promise<void> {
+    const throughRepository = await this.getThroughRepository();
+    const throughConstraint = this.getThroughConstraintOnSource();
+    const targetConstraint = this.getThroughConstraintOnTarget([targetId]);
+    const constraints = {...targetConstraint, ...throughConstraint};
+    await throughRepository.create(
+      constrainDataObject(
+        options?.throughData ?? {},
+        constraints as DataObject<ThroughEntity>,
+      ),
+      options?.throughOptions,
+    );
   }
 
   async unlink(
-    targetModelId: TargetID,
+    targetId: TargetID,
     options?: Options & {
       throughOptions?: Options;
     },
   ): Promise<void> {
-    throw new Error('Method not implemented.');
+    const throughRepository = await this.getThroughRepository();
+    const throughConstraint = this.getThroughConstraintOnSource();
+    const targetConstraint = this.getThroughConstraintOnTarget([targetId]);
+    const constraints = {...targetConstraint, ...throughConstraint};
```

Comment:
Is it safe to combine constraints this way? Should we use WhereBuilder instead to ensure complex syntax like `or` is handled correctly?

```suggestion
    const constraints = new WhereBuilder(targetConstraint).and(throughConstraint).build();
```

@raymondfeng you may have better insight into what's safe and what's not, could you PTAL?